src/infrastructure/admin.py

```diff
@@ -36,32 +45,171 @@ class TechnologyAdmin(admin.ModelAdmin):
+        queryset=User.objects.all(), required=False, label="Select User"
+    )
+
+
+class SemicolonCSV(TextFormat):
+    TABLIB_MODULE = "tablib.formats._csv"
+    CONTENT_TYPE = "text/csv"
+
+    def create_dataset(self, in_stream, **kwargs):
+        dataset = super().create_dataset(in_stream, **kwargs)
+        dataset = tablib.Dataset().load(in_stream, format="csv", delimiter=";")
+        return dataset
+
+    def export_data(self, dataset, **kwargs):
+        return dataset.export("csv", delimiter=";")
+
+
 @admin.register(Project)
-class ProjectAdmin(admin.ModelAdmin):
+class ProjectAdmin(ImportExportModelAdmin):
+    resource_class = ProjectResource
     list_display = ("title", "created_at", "updated_at", "user")
     list_filter = ("title", "created_at", "updated_at", "user")
     search_fields = ("title", "description")
     ordering = ("title",)
     filter_horizontal = ("industries", "technologies")
+    readonly_fields = ("created_at", "updated_at")
+
+    def get_export_formats(self):
+        formats = super().get_export_formats()
+        for i, format_class in enumerate(formats):
+            if format_class is CSV:
+                formats[i] = SemicolonCSV
+        return formats
 
-    fieldsets = (
-        (
-            None,
-            {
-                "fields": (
-                    "title",
-                    "description",
-                    "image",
-                    "user",
-                    "industries",
-                    "technologies",
+    def get_import_formats(self):
+        formats = super().get_import_formats()
+        for i, format_class in enumerate(formats):
+            if format_class is CSV:
+                formats[i] = SemicolonCSV
+        return formats
+
+    def import_action(self, request, *args, **kwargs):
+        user_form = UserSelectForm(formats=[CSV], resources=[ProjectResource])
+
+        if request.method == "POST":
+            user_form = UserSelectForm(
+                formats=[CSV],
+                files=request.FILES,
+                resources=[ProjectResource],
+                data={"user": request.POST.get("user"), "format": "0"},
+            )
+
+            if user_form.is_valid():
+                file = request.FILES["import_file"]
+                data = file.read().decode("utf-8")
+
+                dataset = self.get_import_formats()[0]().create_dataset(
+                    data, delimiter=";"
                 )
-            },
-        ),
-        (
-            "Timestamps",
-            {"fields": ("created_at", "updated_at"), "classes": ("collapse",)},
-        ),
-    )
+                result = self.process_dataset(
+                    dataset,
+                    request=request,
+                    form=user_form,
+                    user_id=user_form.cleaned_data["user"].id,
+                    *args,
+                    **kwargs,
+                )
+                print(f"Result: {result.totals}")
```

Comment:
remove print if we don't need it here

Reply:
i would like to keep this print in order to change "print" -> "log" in future. This logs is gonna be quite important for import/export functionality
@DenysDemchenkoPlaneks